packages/smooth_app/test/utils/mocks.dart

```diff
@@ -0,0 +1,150 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+import 'dart:typed_data';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:mockito/mockito.dart';
+import 'package:provider/provider.dart';
+import 'package:smooth_app/data_models/product_preferences.dart';
+import 'package:smooth_app/data_models/user_preferences.dart';
+import 'package:smooth_app/themes/smooth_theme.dart';
+import 'package:smooth_app/themes/theme_provider.dart';
+
+/// A wrapper for testing various pages of the app with a simple state.
+class MockSmoothApp extends StatelessWidget {
+  const MockSmoothApp(
+    this.userPreferences,
+    this.productPreferences,
+    this.themeProvider,
+    this.child,
+  );
+
+  final UserPreferences userPreferences;
+  final ProductPreferences productPreferences;
+  final ThemeProvider themeProvider;
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    return MultiProvider(
+      providers: <ChangeNotifierProvider<dynamic>>[
+        ChangeNotifierProvider<UserPreferences>.value(value: userPreferences),
+        ChangeNotifierProvider<ProductPreferences>.value(value: productPreferences),
+        ChangeNotifierProvider<ThemeProvider>.value(value: themeProvider),
+      ],
+      child: MaterialApp(
+        localizationsDelegates: AppLocalizations.localizationsDelegates,
+        supportedLocales: AppLocalizations.supportedLocales,
+        theme: SmoothTheme.getThemeData(
+          Brightness.light,
+          themeProvider.colorTag,
+        ),
+        darkTheme: SmoothTheme.getThemeData(
+          Brightness.dark,
+          themeProvider.colorTag,
+        ),
+        themeMode: themeProvider.darkTheme
+          ? ThemeMode.dark
+          : ThemeMode.light,
+        home: child,
+      )
+    );
+  }
+}
+
+Map<String, Object> mockSharedPreferences({
+  String colorTag = 'blue',
+  bool init = true,
+  bool themeDark = false,
+}) => <String, Object>{
+  'IMPORTANCE_AS_STRINGallergens_no_fish' : 'not_important',
```

Comment:
nit: Should these items be sorted in some meaningful way?

Reply:
This is the order they came back as in the real app when I investigated this, I don't know that the order matters so I'll see if I can make it look nicer.